caffe2/CMakeLists.txt

```diff
@@ -211,9 +206,596 @@ if(NOT BUILD_ATEN_ONLY)
+    )
+    if (USE_CUDA AND NOT USE_ROCM)
+      list(APPEND Caffe2_GPU_SRCS
+        ${TORCH_SRC_DIR}/csrc/jit/fuser/cuda/fused_kernel.cpp
+      )
+      add_library(thnvrtc SHARED ${TORCH_SRC_DIR}/csrc/jit/fuser/cuda/thnvrtc.cpp)
+      target_link_libraries(thnvrtc ${CUDA_NVRTC} ${CUDA_CUDA_LIB} ${CUDA_NVRTC_LIB})
+      target_include_directories(thnvrtc PRIVATE ${CUDA_INCLUDE_DIRS})
+      install(TARGETS thnvrtc DESTINATION "${TORCH_INSTALL_LIB_DIR}")
+
+    endif()
+  endif ()
+
+  if (USE_CUDA)
+    list(APPEND Caffe2_GPU_SRCS
+      ${TORCH_SRC_DIR}/csrc/autograd/profiler_cuda.cpp
+      ${TORCH_SRC_DIR}/csrc/autograd/functions/comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
+    )
+  endif()
+
+  if (USE_ROCM)
+    list(APPEND Caffe2_HIP_SRCS
+      ${TORCH_SRC_DIR}/csrc/autograd/functions/comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
+    )
+  endif()
+
+  if (NOT NO_API)
+    list(APPEND TORCH_SRCS
+      ${TORCH_SRC_DIR}/csrc/api/src/cuda.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/datasets/mnist.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/distributed.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/random.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/sequential.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/stream.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/jit.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/init.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/module.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/batchnorm.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/conv.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/dropout.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/embedding.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/functional.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/linear.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/named_any.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/rnn.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/adagrad.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/adam.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/lbfgs.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/optimizer.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/rmsprop.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/serialize.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/sgd.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/serialize/input-archive.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/serialize/output-archive.cpp
+    )
+  endif()
+
+
+  list(APPEND Caffe2_CPU_SRCS ${TORCH_SRCS})
+endif()
+
+# ==========================================================
+# END formerly-libtorch sources
+# ==========================================================
+
+
+
 # Compile exposed libraries.
 add_library(caffe2 ${Caffe2_CPU_SRCS})
+
+
+# Wrapper library for transition to merged libcaffe and libtorch.
+# Only necessary on Windows?
+# Contains "caffe2" and "caffe2_gpu".
+file(WRITE ${CMAKE_BINARY_DIR}/empty.cpp "")
+add_library(torch STATIC ${CMAKE_BINARY_DIR}/empty.cpp)
```

Comment:
What does this solve on windows? It looks like torch is just a wrapper around caffe2 and caffe2_gpu; what do we need this target for?